extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/device/EDS006x.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal.device;
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.*;
+
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Illuminance;
+import javax.measure.quantity.Pressure;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.Util;
+import org.eclipse.smarthome.binding.onewire.internal.handler.OwBaseBridgeHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.OwBaseThingHandler;
+import org.eclipse.smarthome.binding.onewire.internal.owserver.OwserverDeviceParameter;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.MetricPrefix;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EDS006x} class defines an EDS006x device
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class EDS006x extends AbstractOwDevice {
+    private final Logger logger = LoggerFactory.getLogger(EDS006x.class);
+
+    private final OwDeviceParameterMap temperatureParameter = new OwDeviceParameterMap() {
+        {
+            set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/temperature"));
+        }
+    };
+
+    private final OwDeviceParameterMap humidityParameter = new OwDeviceParameterMap() {
+        {
+            set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/humidity"));
+        }
+    };
+
+    private final OwDeviceParameterMap pressureParameter = new OwDeviceParameterMap() {
+        {
+            set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/pressure"));
+        }
+    };
+
+    private final OwDeviceParameterMap lightParameter = new OwDeviceParameterMap() {
+        {
+            set(THING_TYPE_OWSERVER, new OwserverDeviceParameter("/light"));
+        }
+    };
+
+    public EDS006x(String sensorId, OwBaseThingHandler callback) {
+        super(sensorId, callback);
+    }
+
+    @Override
+    public void configureChannels() {
+        isConfigured = false;
+    }
+
+    /**
+     * configure channels for EDS sensors
+     *
+     * @param sensorType an OwSensorType
+     */
+    public void configureChannels(OwSensorType sensorType) {
+        temperatureParameter.set(THING_TYPE_OWSERVER,
```

Comment:
We could with a line `final String sensorTypeName = sensorType.name()` and use `sensorTypeName` instead of calling `sensorType.name()` four times.